### PR TITLE
BHBC 1955: Summary Templates

### PR DESCRIPTION
**Pull request by AlfredRosenthal**
# Overview

## Links to Jira tickets

[BHBC-1955](https://quartech.atlassian.net/browse/BHBC-1955?atlOrigin=eyJpIjoiM2I2Y2JmMzkxNGVkNGZjZTg2ODZmZmU3ZTg0ZGJjYjQiLCJwIjoiaiJ9)

## Description of relevant changes

- updated displayed errors
- fixed sql fetching summary submission status

## Testing
- PR Pod currently has all templates uploaded
- Download the 2.0 version Summary Templates from the resources page 
- Create/ Open a project
- Create/ Open a Survey
- Edit survey to have desired species (Moose, Elk, Sheep, Goat or Caribou)
- upload species summary template to survey

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/src/repositories/summary-repository.ts [hunk 1]
```diff
@@ -399,6 +400,17 @@ export class SummaryRepository extends BaseRepository {
   ): Promise<ISummaryTemplateSpeciesData[]> {
     const templateRow = await this.getSummaryTemplateIdFromNameVersion(templateName, templateVersion);
 
+    if (!templateRow) {
```
Comment (NickPhura) [inline api/src/repositories/summary-repository.ts:412]: nit: Further down we throw an error (line 426). In the `getTemplateMethodologySpeciesRecord` function we do a similar thing, and they create the error as a variable and throw it in both places. Maybe we can do the same here.

Comment (AlfredRosenthal) [reply api/src/repositories/summary-repository.ts:412]: good to keep that consistent, it has been fixed

**Comment by NickPhura** [comment]
@AlfredRosenthal where are we storing the configs for this (wanted to install them locally)?

**Comment by AlfredRosenthal** [comment]
> @AlfredRosenthal where are we storing the configs for this (wanted to install them locally)?
The inserts/ update SQL can be found in confluence as well as a branch in the utils repository

**Comment by NickPhura** [comment]
@AlfredRosenthal Should validation errors be displayed? I get the message "Resolve the following errors in your local file and re-import." but no other errors are displayed. Unless there is some work needed there that isnt part of this ticket?

I assume the status/message display is pretty out of date, given the recent updates to observations. So probably just need to create a ticket for this if we dont have one already.

**Review by NickPhura — APPROVED** [review]
Tried Caribou locally, happy/sad paths

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**